api/examples/prover_run.rs

```diff
@@ -40,7 +40,14 @@ fn main() {
     println!("<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<");
 
     println!("Proving execution of length {}...", trace.blocks.len());
-    let proof = prover::prove::prove_seq(&public_params, trace).expect("error proving execution");
+    let tr = prover::nova::init_circuit_trace(trace).expect("error constructing circuit");
+
+    let mut proof =
+        prover::nova::prove_seq_step(None, &public_params, &tr).expect("error proving execution");
+    for _ in 1..tr.steps() {
+        proof = prover::nova::prove_seq_step(Some(proof), &public_params, &tr)
+            .expect("error proving execution");
+    }
```

Comment:
Is there perhaps an off-by-one error here?

Since `..` is exclusive at the end, this executes one time less than the number of blocks in the trace. If we want one-call per block you could change the iterator's start value from `1` to `0`. Alternatively, there's `..=` which is inclusive at the end.

Reply:
The initial invocation at L45 handles the first step.

This function call should not be user facing (it'll be hidden behind the SDK and the CLI), but thinking about it it's probably worth introducing a more misuse resistant variant for when we don't need to hook in telemetry (like here), so I'll add that.